### DISCUSSION TITLE
Add full Sharp option support

### DISCUSSION THREAD

**Discussion by jmikrut** [payloadcms/payload#443] [Feature Requests & Ideas]
Right now, when resizing images, we only support the following Sharp options:

- width
- height
- crop

But we could pretty easily add support for all other options too.

We should extend Payload to allow for _all_ Sharp image processing options.

**Comment by alexbechmann** [comment]
@jmikrut It would be really nice if dynamic image processing was also possible, so the "sizes" in the collection config become the named crops, but you can still get a crop for another size from the frontend if it's not already predefined e.g.:

```
http://localhost:3000/media/my_image_id?width=100&height=100
```

**Reply by jmikrut** [reply]
This would be a great feature request. We could open up an additional endpoint mounted on the collection router if `upload` is enabled for that collection.

We would accept a PR for this! Not sure when we would be able to get to it but I think this would indeed be a good addition.

**Reply by alexbechmann** [reply]
Hi @jmikrut - I made a draft PR with this: <https://github.com/payloadcms/payload/pull/851>

**Comment by falko100** [comment]
An addition to this would be to enable additional output to `.webp` to improve file size of the images and make Google happy (web core vitals).

**Comment by JosefBredereck** [comment]
I would like to see this addition. The `strategy` option is valuable to have faces always in place and not halfway cutoff.
